### DISCUSSION TITLE
How to join by custom properties

### DISCUSSION THREAD

**Discussion by nonamemanah** [LGouellec/streamiz#305] [Q&A]
Hello! I need your help.
I have two topics. There is json content in each topics. I need merge messages by unique property in json.
For example:
First json customer:
```json
{
  "customer_id":  102,
  "first_name": "noname",
  "age": 38
}
```

second json:
```json
{
  "order_id": 1,
  "custId": 102
}
```

I don't using any dto in my code. Can use JsonObject. I need usage custom properties customer_id and custId as relation properties for merge. Result message will be send to third topic if first and second messages linked by this properties How can I usage this two properties as relation keys?
Thx!

**Comment by LGouellec** [comment]
Hi @nonamemanah , 

I don't know if you solved this issue, but you probably have to change the key in your topology like this sample : 

``` csharp
var stream1 = builder.Stream<String, Customer>("customers"); // key is customerId
var stream2 = builder.Stream<String, Order>("orders"); // key is orderId

stream2
.SelectKey((k,v) => v.CustId)
.Join(stream1, (order, customer) => .... );

```

Hope it's help
